main.go

```diff
@@ -97,8 +187,17 @@ func main() {
 
 	output, err := json.MarshalIndent(&extractedCred, "", "  ")
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("Unable to marshal extracted credential: %v", err)
 	}
 
 	fmt.Print(string(output))
+
+	// update cache
+	if cache_dir != "" {
+		if cache, err := NewCache(cache_dir); err == nil {
```

Comment:
if some error happens it is silenced , change test for err != null

Reply:
Fixed.